From 29a9296cd68516e76d0bd8da320754a222c4ee45 Mon Sep 17 00:00:00 2001
From: Dusty Mabe <dusty@dustymabe.com>
Date: Wed, 13 Sep 2017 11:08:56 -0400
Subject: [PATCH 1/2] resizefs: pass mount point to xfs_growfs

Supposedly it was never a feature to be able to pass a path to a block
device to xfs_growfs and have it grow the filesystem. The behavior changed
upstream recently. It is only supported to pass the mount point of a mounted
XFS filesystem. This causes breakages in cloud-init.

Upstream xfs change was commit b97815a0321072a7154ecab63e297af84066fc78.
 https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/commit/?id=b97815a0321

rhbz: rhbz: https://bugzilla.redhat.com/show_bug.cgi?id=1490505

Signed-off-by: Dusty Mabe <dusty@dustymabe.com>
---
 cloudinit/config/cc_resizefs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cloudinit/config/cc_resizefs.py b/cloudinit/config/cc_resizefs.py
index f14d383..f42b6a6 100644
--- a/cloudinit/config/cc_resizefs.py
+++ b/cloudinit/config/cc_resizefs.py
@@ -67,7 +67,7 @@ def _resize_ext(mount_point, devpth):
 
 
 def _resize_xfs(mount_point, devpth):
-    return ('xfs_growfs', devpth)
+    return ('xfs_growfs', mount_point)
 
 
 def _resize_ufs(mount_point, devpth):
-- 
2.9.4

